packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -545,13 +560,24 @@ class PageScrollPhysics extends ScrollPhysics {
     return page * position.viewportDimension;
   }
 
-  double _getTargetPixels(ScrollMetrics position, Tolerance tolerance, double velocity) {
-    double page = _getPage(position);
+  double _getTargetPage(double page, Tolerance tolerance, double velocity) {
     if (velocity < -tolerance.velocity)
       page -= 0.5;
     else if (velocity > tolerance.velocity)
       page += 0.5;
-    return _getPixels(position, page.roundToDouble());
+    return page.roundToDouble();
+  }
+
+  double _getPageDelta(ScrollMetrics position, Tolerance tolerance, double velocity) {
+    final double page = _getPage(position);
+    final double targetPage = _getTargetPage(page, tolerance, velocity);
+    return targetPage - page;
+  }
+
+  double _getTargetPixels(ScrollMetrics position, Tolerance tolerance, double velocity) {
```

Comment:
@Piinks I just thought whether would it make sense to make some of these methods public, so people can restore the old implementation if they want (or just change it to whatever else)?

Reply:
I don't know if that would be super beneficial, ultimately the changes you are introducing are the way it should be, there are just a lot of tests out there that expect the old behavior.